pkg/controller/certificaterequests/selfsigned/selfsigned.go

```diff
@@ -147,7 +155,7 @@ func (s *SelfSigned) Sign(ctx context.Context, cr *v1alpha1.CertificateRequest,
 	}
 
 	// sign and encode the certificate
-	certPem, _, err := pki.SignCertificate(template, template, publickey, privatekey)
+	certPem, _, err := s.signingFn(template, template, publickey, privatekey)
```

Comment:
So here we mock `SignCertificate` instead of `GenerateTemplate`... can you make the PKI related mocking a bit more consistent? (as discussed on Slack)

Reply:
The issue here is that its difficult to roll generate template and the signing func into one since they're not together like in the CA controller. We could _also_ template out generateTemplateFromCertificateRequest like we do in CA